### ISSUE TITLE
Streamlit: SQL compilation error: Cannot create a Python function with the specified packages

### ISSUE THREAD

**Issue by Unperterbed** [Snowflake-Labs/sfquickstarts#1669]
**Describe the bug**
When deploying the Streamlit app from the RAG quickstart, there's a package conflict with snowflake-snowpark-python and the Python version.

Here's the error message:
`Error running Streamlit: SQL compilation error: Cannot create a Python function with the specified packages. Please check your packages specification and try again. 'One or more package conflicts were detected.'. Hint: These packages are available in other python runtimes:["snowflake-snowpark-python"->[3.8, 3.9, 3.11]].
`

I got this error using the default account settings, and then manually enabled [bundle 2024_08](https://docs.snowflake.com/en/release-notes/bcr-bundles/2024_08_bundle#label-behavior-change-bundle-2024-08-status) so I could change the Python version used for streamlit apps. Even when switching to other versions (I tried 3.10 and 3.11) I got the same message, but the selected Python version didn't appear in the error message. (e.g., 3.10 isn't in the quote above because that was the active version when I copied the text)

**URL of where you see the bug**
https://quickstarts.snowflake.com/guide/ask_questions_to_your_own_documents_with_snowflake_cortex_search/#4

**Screenshots**
![image](https://github.com/user-attachments/assets/47935597-fd01-422d-b5f4-cff58cb31f1d)

**Desktop (please complete the following information):**
 - OS: Windows 11
 - Browser: Chrome 129.0.6668.90 (Official Build) (64-bit)

**Additional context**

**Comment by ccarrero-sf** [comment]
This is already commented in the quickstart. It recommends to use this combination:

snowflake.core = 0.9.0
python = 3.8
snowflake-snowpark-python = 1.22.1
streamlit = 1.26.0

Hope that helps.

**Comment by Unperterbed** [comment]
> This is already commented in the quickstart. It recommends to use this combination:
> 
> snowflake.core = 0.9.0 python = 3.8 snowflake-snowpark-python = 1.22.1 streamlit = 1.26.0
> 
> Hope that helps.

That definitely helps! It's works with those settings, but I don't see them listed anywhere in the quickstart, even now as I'm specifically looking for them.

Several images aren't loading for me ([reported in this issue](https://github.com/Snowflake-Labs/sfquickstarts/issues/1577)), so perhaps that's why I missed it?
![image](https://github.com/user-attachments/assets/5b30c1a3-fe8c-4e27-859c-d1f50d21b648)

**Comment by iamontheinet** [comment]
Can you not see images on this page [https://quickstarts.snowflake.com/guide/ask_questions_to_your_own_documents_with_snowflake_cortex_search/index.html?index=..%2F..index#4](https://quickstarts.snowflake.com/guide/ask_questions_to_your_own_documents_with_snowflake_cortex_search/index.html?index=..%2F..index#4)?

**Comment by Unperterbed** [comment]
I....I....I don't know what to say. The images loaded right away when I clicked on the link you shared. I'd only seen a few of them before, as many did not load after switching between pages dozens of times, and the computer going in/out of sleep mode (though perhaps never a full page refresh). Now the images are loading on every page, and I can't reproduce it even after deleting the browser cache etc and using incognito.
Glad I posted a screenshot of it in my first comment, as proof (even if just for myself) that I'm not crazy!
